.github/workflows/master-sync.yml

```diff
@@ -0,0 +1,31 @@
+name: Synchronize Development & Production
+on:
+  push:
+    branches:
+      - master
+env:
+  AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+  AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+  AWS_DEFAULT_REGION: 'us-east-1'
+jobs:
+  sync_algolia_settings:
+    runs-on: ubuntu-18.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Setup Development Environment Variables
+        run: |
+          echo "ALGOLIA_APP_ID=$(aws ssm get-parameter --name 'algolia-app-id-dev' --query Parameter.Value --output text)" >> $GITHUB_ENV
+          echo "ALGOLIA_API_KEY=$(aws ssm get-parameter --name 'algolia-api-key-ci-dev' --query Parameter.Value --output text)" >> $GITHUB_ENV
+          echo "::add-mask::$(aws ssm get-parameter --name 'algolia-api-key-ci-dev' --query Parameter.Value --output text)"
+          echo "ALGOLIA_RESEARCH_OUTPUT_INDEX=asap-hub_research_outputs_dev >> $GITHUB_ENV
+      - name: Synchronize Development Schema
+        run: yarn run algolia:upload
+      - name: Setup Production Environment Variables
```

Comment:
I'd slit this into two jobs, the prod one would have the `needs: sync_algolia_settings_dev` config